### PR TITLE
Feat/upgrade gdpr to svelte 5

### PR DESCRIPTION
**Pull request by serafim-san**
## Summary
Migrate GDPR component to Svelte 5

## Notion card
https://www.notion.so/santiment/Migrate-Gdpr-to-Svelte-5-13b2a82d1361803d8672f468751d3568

## Screenshots
image
image

### CODE REVIEW

File: src/stories/Design System - App UI/Gdpr/index.svelte [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  import { Gdpr } from '$lib/ui/app/LoginPrompt/index.js'
+
+  const currentUser = useCustomerCtx()
+
+  function onAccept(username: string) {
+    if (currentUser.currentUser.$$) {
+      currentUser.currentUser.$$.privacyPolicyAccepted = true
+    }
+
+    if (window.onGdprAccept) window.onGdprAccept()
```
Comment (serafim-san) [inline src/stories/Design System - App UI/Gdpr/index.svelte:13]: I moved the analytics logic and policy update outside, removing side effects from the component for better readability and reusability

File: src/lib/analytics/events/onboarding.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { trackEvent } from './index.js'
+
+export const trackGdprAccept = (accepted: boolean) =>
+  trackEvent('gdpr_accept', {
+    accepted,
+    category: 'Onboarding',
+    source_url: window.location.href,
```
Comment (DmitriVanGuard) [inline src/lib/analytics/events/onboarding.ts:7]:
```suggestion
```

File: src/lib/utils/fn.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  const clear = () => BROWSER && window.clearTimeout(timer)
+  const debounced = (...args: any) => {
+    clear()
+    return new Promise((resolve) => {
+      timer = window.setTimeout(() => resolve(fn(...args)), timeout)
+    })
+  }
+
+  return [debounced, clear] as [(...args: Parameters<T>) => Promise<ReturnType<T>>, () => void]
+}
```
Comment (DmitriVanGuard) [inline src/lib/utils/fn.ts:15]: Use rxjs debounce in this case

File: src/stories/Design System - App UI/LoginForm/Gdpr.svelte [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  function onAccept(username: string) {
+    console.log('username', username)
+  }
+</script>
+
+<div class="flex h-screen items-center justify-center">
+  {#if !currentUser.currentUser.$$ || currentUser.currentUser.$$.privacyPolicyAccepted}
+    <h1 class="text-base">Gdpr policy is accepted</h1>
+  {:else}
+    <Gdpr currentUser={currentUser.currentUser.$$} {onAccept} />
```
Comment (DmitriVanGuard) [inline src/stories/Design System - App UI/LoginForm/Gdpr.svelte:16]: Gdpr should use customer ctx inside its component

File: src/lib/ui/app/LoginForm/api.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+export const mutateChangeUsername = ApiMutation((variables: { username: string }) => ({
+  schema: `mutation($username: String!) {
+    changeUsername(username: $username) {
+      id
+    }
+  }
+`,
+  variables,
+}))
```
Comment (DmitriVanGuard) [inline src/lib/ui/app/LoginForm/api.ts:21]: Pass values directly to schema

File: src/lib/ui/app/LoginForm/Gdpr.svelte [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+    if (isDisabled || !username) return
+
+    loading = true
+    const usernamePromise = currentUser?.username
+      ? Promise.resolve()
+      : mutateChangeUsername(Query)({ username })
+
+    usernamePromise
+      .catch(onUsernameChangeError)
+      .then(() => mutateGdpr(Query)({ privacyPolicyAccepted: true }))
```
Comment (DmitriVanGuard) [inline src/lib/ui/app/LoginForm/Gdpr.svelte:68]: Failed username change should cancel whole promise chain.

File: src/lib/ui/app/LoginForm/Gdpr.svelte [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+      .then(() => {
+        currentUser.privacyPolicyAccepted = true
```
Comment (DmitriVanGuard) [inline src/lib/ui/app/LoginForm/Gdpr.svelte:70]: Reload customer instead of direct mutation

**Review by DmitriVanGuard — CHANGES_REQUESTED** [review]

**Review by DmitriVanGuard — APPROVED** [review]

**Pull request merged**